### ISSUE TITLE
Unable to extend notebook/templates/page.html

### ISSUE THREAD

**Issue by elgalu** [jupyterhub/jupyterhub#1784]
Hi, I've successfully set `c.NotebookApp.extra_template_paths` via `jupyter_notebook_config.py`.

I'm able to extend any classic notebook page, e.g.

- tree.html via `{% extends "templates/tree.html" %}`
- edit.html via `{% extends "templates/edit.html" %}`
- etc...

However any extensions made to [page.html](https://github.com/jupyter/notebook/blob/master/notebook/templates/page.html) doesn't affect any page.

I suspect it has to do with this:
https://github.com/jupyterhub/jupyterhub/blob/d962e8bcbcb23e21f8c5292092a2c0b04662d5d9/jupyterhub/singleuser.py#L480-L482

**Comment by minrk** [comment]
Interesting! Need to figure out how to get our additions into the page template without stepping on yours.

**Comment by elgalu** [comment]
Please don't worry about my fork, I'm doing some Jinja magic in the meantime (not using my fork anymore).

**Comment by KGHustad** [comment]
We're running into this issue with our deployment at the University of Oslo. We would like to modify the top bar with links to our docs covering various details specific to our deployment. (Resource quotas, course directories etc.) If we modify `page.html`, the changes are only reflected on the pages specific to JupyterHub (`admin.html`, `home.html`, `spawn.html`, etc.).

It would be nice if at least the top bar could be customized by extending `page.html` as follows:
```
{% extends 'page.html' %}

{% block nav_bar_right_items %}
<!-- Links to documentation go here -->
<!-- After this comes the logout button -->
{{ super() }}
{% endblock %}
```

As a workaround, @minrk suggested extending the individual Jupyter pages, `notebook.html`, `tree.html` etc., which I haven't had the time to try out yet.

**Comment by flavionc** [comment]
> We're running into this issue with our deployment at the University of Oslo. We would like to modify the top bar with links to our docs covering various details specific to our deployment. (Resource quotas, course directories etc.) If we modify `page.html`, the changes are only reflected on the pages specific to JupyterHub (`admin.html`, `home.html`, `spawn.html`, etc.).
> 
> It would be nice if at least the top bar could be customized by extending `page.html` as follows:
> 
> ```
> {% extends 'page.html' %}
> 
> {% block nav_bar_right_items %}
> 
> 
> {{ super() }}
> {% endblock %}
> ```
> 
> As a workaround, @minrk suggested extending the individual Jupyter pages, `notebook.html`, `tree.html` etc., which I haven't had the time to try out yet.

Have you found the solution? This is exactly what i need to do.

**Comment by KGHustad** [comment]
@flavionc No, I never got around to try out the workaround that Min proposed.

**Comment by LukasMueller187** [comment]
Sadly this is still not possible. This did cost me a few hours just to figure out that it is not possible to extend page.html. This should be mentioned on the wiki page.
Would appreciate a fix!
